<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4. Maximum likelihood phylodynamic analysis | IDE Module 5 Lab</title>
  <meta name="description" content="Phylodynamics of SARS-CoV-2 B.1.1.7" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="4. Maximum likelihood phylodynamic analysis | IDE Module 5 Lab" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bioinformaticsdotca.github.io/PGE_2025/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Phylodynamics of SARS-CoV-2 B.1.1.7" />
  <meta name="github-repo" content="bioinformaticsdotca/PGE_2025" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4. Maximum likelihood phylodynamic analysis | IDE Module 5 Lab" />
  
  <meta name="twitter:description" content="Phylodynamics of SARS-CoV-2 B.1.1.7" />
  <meta name="twitter:image" content="https://bioinformaticsdotca.github.io/PGE_2025/img/bioinformatics_logo.png" />

<meta name="author" content="Angela McLaughlin and Finlay Maguire" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="exercise-setup.html"/>
<link rel="next" href="visualizing-the-phylodynamic-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "name": "Pathogen Genomic Epidemiology",
  "description": "The complete set of tutorials, slides, and practical exercises for the Pathogen Genomic Epidemiology 2025 Canadian Bioinformatics Workshop, hosted on GitHub Pages.",
  "url": "https://bioinformaticsdotca.github.io/PGE_2025",
  "learningResourceType": ["tutorial", "slides", "exercise"],
  "inLanguage": "en",
  "license": "https://creativecommons.org/licenses/by/4.0/",
  "provider": {
    "@type": "Organization",
    "name": "Bioinformatics.ca",
    "url": "https://bioinformatics.ca"
  },
  "keywords": "",
  "isPartOf": {
    "@type": "Course",
    "@id": "https://bioinformatics.ca/workshops-all/2025-pathogen-genomic-epidemiology-online/"
  }
}
</script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br><br><br></a></strong></center></li>
<a href="./">
 <img src="img/pge-icon.png" width="70%" alt=" logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#class-photo"><i class="fa fa-check"></i>Class Photo</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="meet-your-faculty.html"><a href="meet-your-faculty.html"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
<li class="chapter" data-level="" data-path="data-and-compute-setup.html"><a href="data-and-compute-setup.html"><i class="fa fa-check"></i>Data and Compute Setup</a></li>
<li class="chapter" data-level="" data-path="module-1-data-curation-and-data-sharing.html"><a href="module-1-data-curation-and-data-sharing.html"><i class="fa fa-check"></i>Module 1 Data Curation and Data Sharing</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-data-curation-and-data-sharing.html"><a href="module-1-data-curation-and-data-sharing.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-1-data-curation-and-data-sharing.html"><a href="module-1-data-curation-and-data-sharing.html#lab"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html"><i class="fa fa-check"></i>Module 2 Emerging Pathogen Detection and Identification</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#lab-1"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#intro"><i class="fa fa-check"></i>1. Introduction</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#software"><i class="fa fa-check"></i>2. List of software for tutorial and its respective documentation</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#setup"><i class="fa fa-check"></i>3. Exercise setup</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#copy-data-files"><i class="fa fa-check"></i>3.1. Copy data files</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#activate-environment"><i class="fa fa-check"></i>3.2. Activate environment</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#verify-your-workshop-machine-url"><i class="fa fa-check"></i>3.3. Verify your workshop machine URL</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#exercise"><i class="fa fa-check"></i>4. Exercise</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#exercise-background"><i class="fa fa-check"></i>4.1. Patient Background:</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#exercise-overview"><i class="fa fa-check"></i>4.2. Overview</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#assembly-free"><i class="fa fa-check"></i>Assembly-free approach</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#assembly-based"><i class="fa fa-check"></i>Assembly-based approach</a></li>
<li class="chapter" data-level="" data-path="module-2-emerging-pathogen-detection-and-identification.html"><a href="module-2-emerging-pathogen-detection-and-identification.html#final"><i class="fa fa-check"></i>5. Final words</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-pathogen-typing.html"><a href="module-3-pathogen-typing.html"><i class="fa fa-check"></i>Module 3 Pathogen Typing</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-pathogen-typing.html"><a href="module-3-pathogen-typing.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3-pathogen-typing.html"><a href="module-3-pathogen-typing.html#lab-2"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-pathogen-typing.html"><a href="module-3-pathogen-typing.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-3-pathogen-typing.html"><a href="module-3-pathogen-typing.html#part-one-chewbbaca-objective"><i class="fa fa-check"></i>PART ONE: chewBBACA Objective</a></li>
<li class="chapter" data-level="" data-path="module-3-pathogen-typing.html"><a href="module-3-pathogen-typing.html#part-two-cgmlst-clustering-objective"><i class="fa fa-check"></i>PART TWO: cgMLST clustering Objective</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html"><i class="fa fa-check"></i>Module 4 Outbreak Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html#lab-3"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html#set-up"><i class="fa fa-check"></i>Set-Up</a></li>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html#cluster-identification"><i class="fa fa-check"></i>Cluster Identification</a></li>
<li class="chapter" data-level="" data-path="module-4-outbreak-analysis.html"><a href="module-4-outbreak-analysis.html#snp-analysis"><i class="fa fa-check"></i>SNP Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-phylodynamics.html"><a href="module-5-phylodynamics.html"><i class="fa fa-check"></i>Module 5 Phylodynamics</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-phylodynamics.html"><a href="module-5-phylodynamics.html#lecture-4"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-5-phylodynamics.html"><a href="module-5-phylodynamics.html#lab-4"><i class="fa fa-check"></i>Lab</a></li>
<li class="chapter" data-level="" data-path="module-5-phylodynamics.html"><a href="module-5-phylodynamics.html#table-of-contents"><i class="fa fa-check"></i>Table of contents</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i>1. Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-1.html"><a href="introduction-1.html#background-1"><i class="fa fa-check"></i>1.1. Background</a></li>
<li class="chapter" data-level="" data-path="introduction-1.html"><a href="introduction-1.html#objectives-of-this-analysis"><i class="fa fa-check"></i>1.2. Objectives of this analysis:</a></li>
<li class="chapter" data-level="" data-path="introduction-1.html"><a href="introduction-1.html#what-statistical-approach-is-best"><i class="fa fa-check"></i>1.3. What statistical approach is best?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="list-of-software-for-tutorial.html"><a href="list-of-software-for-tutorial.html"><i class="fa fa-check"></i>2. List of software for tutorial</a></li>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html"><i class="fa fa-check"></i>3. Exercise setup</a>
<ul>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html#data-source"><i class="fa fa-check"></i>3.0. Data source</a>
<ul>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html#filters-applied-to-download-this-dataset"><i class="fa fa-check"></i>Filters applied to download this dataset</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html#copy-data-files-1"><i class="fa fa-check"></i>3.1. Copy data files</a></li>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html#activate-conda-environment"><i class="fa fa-check"></i>3.2. Activate conda environment</a></li>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html#find-your-ip-address-to-setup-an-aws-instance"><i class="fa fa-check"></i>3.3. Find your IP address to setup an AWS instance</a></li>
<li class="chapter" data-level="" data-path="exercise-setup.html"><a href="exercise-setup.html#inspect-and-clean-data"><i class="fa fa-check"></i>3.4. Inspect and clean data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html"><i class="fa fa-check"></i>4. Maximum likelihood phylodynamic analysis</a>
<ul>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#construct-a-multiple-sequence-alignment-of-the-genomes-augur-align"><i class="fa fa-check"></i>4.1. Construct a multiple sequence alignment of the genomes (<code>augur align</code>)</a></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#index-sequences-augur-index"><i class="fa fa-check"></i>4.2. Index sequences (<code>augur index</code>)</a></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#filter-out-low-quality-sequences-augur-filter"><i class="fa fa-check"></i>4.3. Filter out low quality sequences (<code>augur filter</code>)</a></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#mask-problematic-sites-in-the-alignment"><i class="fa fa-check"></i>4.4. Mask problematic sites in the alignment</a>
<ul>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#optional"><i class="fa fa-check"></i>Optional</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#build-a-maximum-liklihood-phylogenetic-tree-augur-tree"><i class="fa fa-check"></i>4.5. Build a maximum liklihood phylogenetic tree (<code>augur tree</code>)</a></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#inferring-timing-of-host-change-augur-refine"><i class="fa fa-check"></i>4.6. Inferring timing of host change (<code>augur refine</code>)</a>
<ul>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#optional-1"><i class="fa fa-check"></i>Optional</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#infer-ancestral-states-for-host-and-location-augur-traits"><i class="fa fa-check"></i>4.7. Infer ancestral states for host and location (<code>augur traits</code>)</a></li>
<li class="chapter" data-level="" data-path="maximum-likelihood-phylodynamic-analysis.html"><a href="maximum-likelihood-phylodynamic-analysis.html#package-up-data-for-auspice-visualisation-augur-export"><i class="fa fa-check"></i>4.8. Package up data for Auspice visualisation (<code>augur export</code>)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="visualizing-the-phylodynamic-analysis.html"><a href="visualizing-the-phylodynamic-analysis.html"><i class="fa fa-check"></i>5. Visualizing the phylodynamic analysis</a>
<ul>
<li class="chapter" data-level="" data-path="visualizing-the-phylodynamic-analysis.html"><a href="visualizing-the-phylodynamic-analysis.html#load-data-into-auspice"><i class="fa fa-check"></i>5.1. Load data into Auspice</a></li>
<li class="chapter" data-level="" data-path="visualizing-the-phylodynamic-analysis.html"><a href="visualizing-the-phylodynamic-analysis.html#explore-data"><i class="fa fa-check"></i>5.2. Explore data</a></li>
<li class="chapter" data-level="" data-path="visualizing-the-phylodynamic-analysis.html"><a href="visualizing-the-phylodynamic-analysis.html#examine-the-molecular-clock"><i class="fa fa-check"></i>5.3. Examine the molecular clock</a></li>
<li class="chapter" data-level="" data-path="visualizing-the-phylodynamic-analysis.html"><a href="visualizing-the-phylodynamic-analysis.html#examine-the-ancestral-state-reconstruction"><i class="fa fa-check"></i>5.4. Examine the ancestral state reconstruction</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="delphy-analysis.html"><a href="delphy-analysis.html"><i class="fa fa-check"></i>6. Delphy analysis</a>
<ul>
<li class="chapter" data-level="" data-path="delphy-analysis.html"><a href="delphy-analysis.html#modify-fasta-names-then-load-the-renamed-fasta-into-delphy"><i class="fa fa-check"></i>6.1. Modify fasta names, then load the renamed fasta into Delphy</a></li>
<li class="chapter" data-level="" data-path="delphy-analysis.html"><a href="delphy-analysis.html#modify-priors-by-clicking-on-advanced-options"><i class="fa fa-check"></i>6.2. Modify priors by clicking on advanced options</a></li>
<li class="chapter" data-level="" data-path="delphy-analysis.html"><a href="delphy-analysis.html#run-the-analysis-and-interpret-output"><i class="fa fa-check"></i>6.3. Run the analysis and interpret output</a></li>
<li class="chapter" data-level="" data-path="delphy-analysis.html"><a href="delphy-analysis.html#export-results"><i class="fa fa-check"></i>6.4. Export results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="end-of-lab.html"><a href="end-of-lab.html"><i class="fa fa-check"></i>7. End of lab</a></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://www.cihr-irsc.gc.ca"><img src="img/sponsors/cihr.png" width=90% style="padding-bottom: 20px" alt="Canadian Institutes of Health Research logo."></a>

<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="55" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="55" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=35% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
<br><br>
<a href="https://uhn.ca/"><img src="img/sponsors/uhnlogo.png" width=55% style="display:inline-block;" style="padding-bottom: 20px" alt="University Health Network logo.">

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">IDE Module 5 Lab</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="maximum-likelihood-phylodynamic-analysis" class="section level1 hasAnchor">
<h1>4. Maximum likelihood phylodynamic analysis<a href="maximum-likelihood-phylodynamic-analysis.html#maximum-likelihood-phylodynamic-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The overall goal of this lab is to estimate where and when B.1.1.7 most likely originated. To do this, we will mostly make use of the <a href="https://docs.nextstrain.org/projects/augur/en/stable/index.html">Augur</a> maximum likelihood tool suite, which powers the <a href="https://nextstrain.org/">NextStrain</a> website. Within augur, the key tool actually being used for most of our inferences is <a href="https://treetime.readthedocs.io/en/latest/">TreeTime</a> a likelihood based tool for inferring molecular clocks and ancestral traits.</p>
<p><strong>An overview of the basic usage of Augur (figure from the Augur documentation):</strong>
<img src="img/PGE_mod5/augur_analysis_sketch.webp" alt="augur workflow" /></p>
<p>The steps/functions we use are in this section of the lab are the following:</p>
<ul>
<li><strong>align</strong>: This step constructs a multiple sequence alignment from this subset of genomes.</li>
<li><strong>index</strong>: This step indexes the sequences, counting the frequency of nucleotides and gaps.</li>
<li><strong>filter</strong>: This step filters out sequences that have too many gaps or ambiguous nucleotides.</li>
<li><strong>tree</strong>: This step builds a phylogenetic tree, where branch lengths are measured in substitutions/site (a divergence tree).</li>
<li><strong>refine</strong>: This step constructs a time tree using our existing tree alongside collection dates of SARS-CoV-2 genomic samples (branch lengths are measured in time).</li>
<li><strong>traits</strong>: This step uses treetime’s mugration model to infer ancestral traits based on the tree. We will use this to infer ancestral host and geography.</li>
<li><strong>export</strong>: This step exports the data to be used by <a href="https://auspice.us/">Auspice</a>, a version of the visualization system used by NextStrain that lets you examine your own phylogenetic tree and metadata.</li>
</ul>
<hr />
<div id="construct-a-multiple-sequence-alignment-of-the-genomes-augur-align" class="section level2 hasAnchor">
<h2>4.1. Construct a multiple sequence alignment of the genomes (<code>augur align</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#construct-a-multiple-sequence-alignment-of-the-genomes-augur-align" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first step is to construct a multiple sequence alignment of the genomes, which is required before building a phylogenetic tree. We will be using the command <code>augur align</code> to accomplish this task, but underneath this <a href="https://mafft.cbrc.jp/alignment/software/">mafft</a> runs to construct the alignment.</p>
<p>To build a new folder and construct the alignment, please run the following:</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb76-1" tabindex="-1"></a><span class="fu">mkdir</span> data/aligned</span>
<span id="cb76-2"><a href="maximum-likelihood-phylodynamic-analysis.html#cb76-2" tabindex="-1"></a></span>
<span id="cb76-3"><a href="maximum-likelihood-phylodynamic-analysis.html#cb76-3" tabindex="-1"></a><span class="ex">augur</span> align <span class="at">--nthreads</span> 2 <span class="at">--sequences</span> data/complete/b117_seq_complete_sampUK.fasta <span class="at">--reference-sequence</span> data/download/MN908947.3.fasta <span class="at">--output</span> data/aligned/b117_seq_complete_sampUK_aligned.fasta</span></code></pre></div>
<p>You should expect to see the following:</p>
<p><strong>Output</strong></p>
<pre><code>using mafft to align via:
    mafft --reorder --anysymbol --nomemsave --adjustdirection --thread 2 data/b117_seq_complete_sampUK_aligned.fasta.to_align.fasta 1&gt; data/b117_seq_complete_sampUK_aligned.fasta 2&gt; data/b117_seq_complete_sampUK_aligned.fasta.log 

    Katoh et al, Nucleic Acid Research, vol 30, issue 14
    https://doi.org/10.1093%2Fnar%2Fgkf436

WARNING: 1bp insertion at ref position 200 was due to &#39;N&#39;s or &#39;?&#39;s in provided sequences
WARNING: 1bp insertion at ref position 6337 was due to &#39;N&#39;s or &#39;?&#39;s in provided sequences
Trimmed gaps in MN908947.3|China|2019-12-26 from the alignment</code></pre>
<p>The meaning of each parameter:</p>
<ul>
<li>`–nthreads: Number of threads for the alignment.</li>
<li>`–sequences: The input set of sequences in FASTA format.</li>
<li>`–output: The output alignment, in FASTA format.</li>
<li>`–reference-sequence: The reference genome (MN908947.3==Wuhan-Hu-1).</li>
</ul>
<p>The reference sequence will be included in our alignment and <code>augur align</code> removes any insertions with respect to this reference genome (useful when identifying and naming specific mutations later on in the augur pipeline).</p>
<p>The aligned fasta file is a very similar format as the input file, but the difference is that sequences have been aligned (possibly by inserting gaps <code>-</code>). This also means that all sequences in <code>alignment.fasta</code> should have the same length (whereas sequences in <code>sequences.fasta</code>, which is not aligned, may have different lengths). When time allows, it is good practice to visually inspect the alignment in a viewer such as <a href="https://github.com/AliView/AliView">AliView</a>.</p>
</div>
<div id="index-sequences-augur-index" class="section level2 hasAnchor">
<h2>4.2. Index sequences (<code>augur index</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#index-sequences-augur-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This step is optional but useful to inspect sequence quality and quicken the subsequent filtering step, by counting the frequency of nucleotides and gaps.</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb78-1" tabindex="-1"></a><span class="ex">augur</span> index <span class="at">--sequences</span> data/aligned/b117_seq_complete_sampUK_aligned.fasta  <span class="at">--output</span>  data/aligned/b117_seq_complete_sampUK_aligned_sequence_index.tsv</span></code></pre></div>
<p>Inspect the index output in the ‘aligned’ subfolder.</p>
<div type="orange" style="regular" title="Questions" collapsible="false" icon="true">
<ul>
<li>Knowing that the genome size is 29,903 bp, do any samples appear to have more gaps or Ns than you would expect?</li>
</ul>
</div>
<div type="green" style="regular" title="Answers" collapsible="true" icon="true">
<ul>
<li>At first glance, none appear to have &gt;~3000 gaps or Ns.</li>
</ul>
</div>
</div>
<div id="filter-out-low-quality-sequences-augur-filter" class="section level2 hasAnchor">
<h2>4.3. Filter out low quality sequences (<code>augur filter</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#filter-out-low-quality-sequences-augur-filter" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Filter seqs to a minimum length (–min-length) of 26913, which represents 90% of 29,903 sites with ACTG (i.e., max 10% ambig and gaps)</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb79-1" tabindex="-1"></a><span class="fu">mkdir</span> data/filtered</span>
<span id="cb79-2"><a href="maximum-likelihood-phylodynamic-analysis.html#cb79-2" tabindex="-1"></a></span>
<span id="cb79-3"><a href="maximum-likelihood-phylodynamic-analysis.html#cb79-3" tabindex="-1"></a><span class="ex">augur</span> filter <span class="at">--nthreads</span> 2 <span class="at">--sequences</span> data/aligned/b117_seq_complete_sampUK_aligned.fasta <span class="at">--metadata</span> data/complete/b117_seq_complete_sampUK.tsv <span class="at">--sequence-index</span> data/aligned/b117_seq_complete_sampUK_aligned_sequence_index.tsv <span class="at">--min-length</span> 26913 <span class="at">--output-sequences</span> data/filtered/b117_seq_complete_sampUK_aligned_filt.fasta <span class="at">--output-metadata</span> data/filtered/b117_seq_complete_sampUK_filt.tsv </span></code></pre></div>
<p><strong>Output</strong></p>
<pre><code>1 strain was dropped during filtering
    1 had no metadata (this is the refence sequence)
178 strains passed all filters
</code></pre>
<p>Nice, all the sequences passed our quality filters (this is rarer than you might think).</p>
<div type="orange" style="regular" title="Questions" collapsible="false" icon="true">
<ul>
<li>Why is it important to remove low quality sequences prior to phylogenetic inference?</li>
<li>The choice of maximum 10% ambiguous and gaps seems (<em>is</em>) somewhat arbitrary. How would increasing or decreasing this threshold change the analysis?</li>
</ul>
</div>
<div type="green" style="regular" title="Answers" collapsible="true" icon="true">
<ul>
<li>Sequences with excessive gaps or ambiguous sites may not have as high of accuracy, and have less signal to contribute to the tree, which could bias its placement.</li>
<li>Increasing the threshold would allow more sequences of lower quality to pass through the filter. Decreasing the threshold is more conservative.</li>
</ul>
</div>
</div>
<div id="mask-problematic-sites-in-the-alignment" class="section level2 hasAnchor">
<h2>4.4. Mask problematic sites in the alignment<a href="maximum-likelihood-phylodynamic-analysis.html#mask-problematic-sites-in-the-alignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Not all sites in the genome are equally informative towards inferring their phylogenetic relationships. Some sites can be misleading and should be masked before building trees. The ends of alignments often exhibit low coverage and a higher rate of sequencing or mapping errors. In SARS-CoV-2, this includes positions 1–55 and 29804–29903. Homoplasic sites at which multiple recurrent mutations arise may reflect contamination, recurrent sequencing errors, or hypermutability. Even if they are accurate, they may not be representative of evolution across the remainder of the genome. Another example (not relevant here) would be sites at which drug resistance mutations occur. In HIV-1 phylodynamic analyses, codons of surveillance drug resistance mutations are masked. For further interest, see this <a href="https://virological.org/t/masking-strategies-for-sars-cov-2-alignments/480">justification for masking SARS-CoV-2 sequences</a>.</p>
<p>Note that augur also has a masking subcommand, but it requires the mask file to be in a different format, so we’re not using it here. Instead, we’re going to run a python script that masks sequences based on a vcf (Variant Call Format) file of problematic sites. In order to run this, we have to concatenate the reference sequence back into the fasta, which was filtered out by augur above. Then we’re going to run our masking script, and then remove that reference again.</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-1" tabindex="-1"></a><span class="fu">mkdir</span> data/masked</span>
<span id="cb81-2"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-2" tabindex="-1"></a></span>
<span id="cb81-3"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-3" tabindex="-1"></a><span class="co">## Concatenate the fasta with the reference fasta</span></span>
<span id="cb81-4"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-4" tabindex="-1"></a><span class="fu">cat</span> data/filtered/b117_seq_complete_sampUK_aligned_filt.fasta data/download/MN908947.3.fasta <span class="op">&gt;</span> data/filtered/b117_seq_complete_sampUK_aligned_filt_ref.fasta</span>
<span id="cb81-5"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-5" tabindex="-1"></a></span>
<span id="cb81-6"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-6" tabindex="-1"></a><span class="co">## Run the masking python script</span></span>
<span id="cb81-7"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-7" tabindex="-1"></a><span class="ex">python3</span> scripts/mask_alignment_using_vcf.py <span class="at">-i</span> data/filtered/b117_seq_complete_sampUK_aligned_filt_ref.fasta <span class="at">-o</span> data/masked/b117_seq_complete_sampUK_aligned_filt_ref_mask.fasta <span class="at">-v</span> scripts/problematic_sites_vcfv5.vcf <span class="at">-n</span> <span class="st">&quot;n&quot;</span> <span class="at">--reference_id</span> <span class="st">&quot;MN908947.3|China|2019-12-26&quot;</span></span>
<span id="cb81-8"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-8" tabindex="-1"></a></span>
<span id="cb81-9"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-9" tabindex="-1"></a><span class="co">## Remove the reference sequence from the alignment using augur filter</span></span>
<span id="cb81-10"><a href="maximum-likelihood-phylodynamic-analysis.html#cb81-10" tabindex="-1"></a><span class="ex">augur</span> filter <span class="at">--sequences</span> data/masked/b117_seq_complete_sampUK_aligned_filt_ref_mask.fasta <span class="at">--metadata</span> data/filtered/b117_seq_complete_sampUK_filt.tsv <span class="at">--output-sequences</span> data/masked/b117_seq_complete_sampUK_aligned_filt_mask.fasta <span class="at">--output-metadata</span> data/masked/b117_seq_complete_sampUK_aligned_filt_mask.tsv</span></code></pre></div>
<p>The $cat command is a simple and useful tool to concatenate multiple fasta files together.</p>
<p>Let’s have a closer look at the parameters we used in the python script:</p>
<ul>
<li><code>-i</code>: Fasta input alignment to be masked (with the ref sequence in it)</li>
<li><code>-o</code>: Fasta output alignment with problematic sites masked</li>
<li><code>-v</code>: The vcf file of problematic sites from <a href="https://virological.org/t/masking-strategies-for-sars-cov-2-alignments/480">de Maio</a></li>
<li><code>-n</code>: the character to replace nucleotides at the masked sites</li>
<li><code>--reference-id</code>: the ID of the reference sequence in the input</li>
</ul>
<div id="optional" class="section level3 hasAnchor">
<h3>Optional<a href="maximum-likelihood-phylodynamic-analysis.html#optional" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Open and inspect the fasta files before and after masking.</li>
</ul>
</div>
</div>
<div id="build-a-maximum-liklihood-phylogenetic-tree-augur-tree" class="section level2 hasAnchor">
<h2>4.5. Build a maximum liklihood phylogenetic tree (<code>augur tree</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#build-a-maximum-liklihood-phylogenetic-tree-augur-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The next step is to take the set of aligned genomes and build a phylogenetic tree (a divergence tree in units of substitutions per site). We will use <code>augur tree</code> for this, but underneath it (by default) runs <a href="http://www.iqtree.org/">iqtree</a>, which uses maximum likelihood to infer a phylogenetic tree.</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb82-1" tabindex="-1"></a><span class="fu">mkdir</span> trees</span>
<span id="cb82-2"><a href="maximum-likelihood-phylodynamic-analysis.html#cb82-2" tabindex="-1"></a></span>
<span id="cb82-3"><a href="maximum-likelihood-phylodynamic-analysis.html#cb82-3" tabindex="-1"></a><span class="ex">augur</span> tree <span class="at">--nthreads</span> 4 <span class="at">--method</span> iqtree <span class="at">--substitution-model</span> GTR <span class="at">--alignment</span> data/masked/b117_seq_complete_sampUK_aligned_filt_mask.fasta <span class="at">--output</span> trees/b117_seq_complete_sampUK_aligned_filt_mask.tre</span></code></pre></div>
<p>Let’s check out the parameters</p>
<ul>
<li><code>--nthreads</code>: Number of threads</li>
<li><code>--method</code>: The tree inference method. We use the default, iqtree(3)</li>
<li><code>--substitution-model</code>: the underlying substitution model, here specified as generalized time reversible (GTR), which means that each nucleotide substitution rate in the matrix (ACTG vs ACTG) is separately estimated.</li>
<li><code>--alignment</code>: the input alignment (with masked sites)</li>
<li><code>--output</code>: the output tree (can be .tre or .nwk files)</li>
</ul>
<p>You should expect to see this (or a very similar) output:</p>
<p><strong>Output</strong></p>
<pre><code>Building a tree via:
    iqtree3 --threads-max 4 -s data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.fasta -m GTR --ninit 2 -n 2 --epsilon 0.05 -T AUTO --redo &gt; data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.iqtree.log
    Please use the corresponding citations according to
    https://iqtree.github.io/doc/Home#how-to-cite-iq-tree

Building original tree took 12.720907926559448 seconds
</code></pre>
<p>This produces as output a <code>.tre</code> file, which is a Newick format phylogenetic tree, which has been midpoint-rooted. You can load this file in a variety of phylogenetic tree viewers (such as <a href="http://phylo.io/" class="uri">http://phylo.io/</a>) but we will further refine this file to work with Auspice.</p>
<p>Another output file is <code>alignment-delim.iqtree.log</code>, which contains additional information from <a href="http://www.iqtree.org/">iqtree</a>. You can take a look at this file to get an idea of what <a href="http://www.iqtree.org/">iqtree</a> was doing by using <code>tail</code> (prints the last few lines of a file).</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb84-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb84-1" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> 20 data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.iqtree.log</span></code></pre></div>
<p><strong>Output</strong></p>
<pre><code>Optimal log-likelihood: -44067.898
Rate parameters:  A-C: 0.24165  A-G: 1.07984  A-T: 0.21863  C-G: 0.31202  C-T: 3.23196  G-T: 1.00000
Base frequencies:  A: 0.299  C: 0.184  G: 0.196  T: 0.321
Parameters optimization took 1 rounds (0.019 sec)
BEST SCORE FOUND : -44067.898
Total tree length: 0.012

Total number of iterations: 2
CPU time used for tree search: 1.277 sec (0h:0m:1s)
Wall-clock time used for tree search: 0.408 sec (0h:0m:0s)
Total CPU time used: 29.925 sec (0h:0m:29s)
Total wall-clock time used: 12.023 sec (0h:0m:12s)

Analysis results written to: 
  IQ-TREE report:                data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.fasta.iqtree
  Maximum-likelihood tree:       data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.fasta.treefile
  Likelihood distances:          data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.fasta.mldist
  Screen log file:               data/masked/b117_seq_complete_sampUK_aligned_filt_mask-delim.fasta.log

Date and Time: Wed Nov 19 15:50:14 2025</code></pre>
<p>As iqtree uses a maximum likelihood approach, you will see that it will report the likelihood score of the optimal tree (reported as log-likelihoods since likelihood values are very small).</p>
<p><em>Note: For this lab we are not looking at branch support values for a tree, but for real-world analysis you may wish to look into including (ultra-fast) bootstrap support values or approximate likelihood ratio test values. This will give a measure of how well-supported each branch in the tree is by the alignment (often a number from 0 for little support to 100 for maximal support). Please see the <a href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with-ultrafast-bootstrap-approximation">IQTree documentation</a> for more details.</em></p>
</div>
<div id="inferring-timing-of-host-change-augur-refine" class="section level2 hasAnchor">
<h2>4.6. Inferring timing of host change (<code>augur refine</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#inferring-timing-of-host-change-augur-refine" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The tree output by <a href="http://www.iqtree.org/">iqtree</a> shows hypothetical evolutionary relationships between different SARS-CoV-2 genomes with branch lengths representing distances between different genomes (in units of <strong>substitutions/site</strong>, i.e., the predicted number of substitutions between genomes divided by the alignment length). However, other methods of measuring distance between genomes are possible. In particular we can incorporate the collection dates of the different SARS-CoV-2 genomes to infer a tree where branches are scaled according to the elapsed time and the dates of internal nodes are inferred. Such trees are called <strong>time trees</strong> or <strong>time-scaled phylogenies</strong>.</p>
<p>We will use <a href="https://treetime.readthedocs.io/en/latest/">TreeTime</a> invoked by <a href="https://docs.nextstrain.org/projects/augur/en/stable/index.html">augur</a> to infer a <strong>time tree</strong> from our phylogenetic tree using collection dates of the SARS-CoV-2 genomes stored in the metadata file. We will use the <code>augur refine</code> step to run TreeTime and perform some additional refinemint of the tree.</p>
<p>To do this, please run the following:</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb86"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb86-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb86-1" tabindex="-1"></a><span class="ex">augur</span> refine <span class="at">--alignment</span> data/masked/b117_seq_complete_sampUK_aligned_filt_mask.fasta <span class="at">--tree</span> trees/b117_seq_complete_sampUK_aligned_filt_mask.tre <span class="at">--metadata</span> data/masked/b117_seq_complete_sampUK_aligned_filt_mask.tsv <span class="at">--timetree</span> <span class="at">--divergence-units</span> mutations <span class="at">--output-tree</span> trees/b117_seq_complete_sampUK_aligned_filt_mask_timetree.tre <span class="at">--output-node-data</span> trees/refine.node.json <span class="at">--seed</span> 123 <span class="at">--stochastic-resolve</span> <span class="at">--root</span> least-squares <span class="at">--date-confidence</span> <span class="at">--clock-rate</span> 4.5e-4 <span class="at">--clock-std-dev</span> 0.2</span></code></pre></div>
<p>Once again, let’s have a closer look at the parameters:</p>
<ul>
<li><code>--alignment</code>: The alignment used to build the tree. Used to re-scale the divergence units.</li>
<li><code>--tree</code>: The starting input tree built using iqtree3.</li>
<li><code>--metadata</code>: The metadata which contains the SARS-CoV-2 genome names (in a column called <code>name</code>) and the sample collection dates (in a column named <code>date</code>).</li>
<li><code>--timetree</code>: Specifiy to build a time tree.</li>
<li><code>--divergence-units</code>: Convert the branch lengths of <strong>substitutions/site</strong> (<strong>mutations/site</strong>) to <strong>mutations</strong> (not needed to build a time tree, this is just used for visualizing the tree later on).</li>
<li><code>--output-tree</code>: The output Newick file containing the time tree.</li>
<li><code>--output-node-data</code>: Augur will store additional information here which will let us convert between time trees and substitution trees.</li>
<li><code>--seed</code>: Set a seed for reproducibility</li>
<li><code>--stochastic-resolve</code>: Resolve polytomies stochastically (forces a bifurcating tree)</li>
<li><code>--root</code>: choose root that optimizes the least squares for the root-to-tip regression
-date-confidence</li>
<li><code>--clock-rate</code>: We are specifying a relaxed molecular clock rate, with a mean of 4.5e-4 based on <a href="https://academic.oup.com/ve/article/8/2/veac080/6677185">Hill et al. 2022</a></li>
<li><code>--clock-std-dev</code>: This is the deviance on the relaxed clock, allowing for variable evolutionary rates across the tree. You could play with this parameter if you have time.</li>
</ul>
<p>You should expect to see the following as output:</p>
<p><strong>Output</strong></p>
<pre><code>augur refine is using TreeTime version 0.11.4

4.28    TreeTime.reroot: with method or node: least-squares

4.28    TreeTime.reroot: rerooting will account for covariance and shared ancestry.

5.32    ###TreeTime.run: INITIAL ROUND

9.24    TreeTime.reroot: with method or node: least-squares

9.24    TreeTime.reroot: rerooting will account for covariance and shared ancestry.

9.41    ###TreeTime.run: rerunning timetree after rerooting

13.59   ###TreeTime.run: ITERATION 1 out of 2 iterations

21.88   ###TreeTime.run: ITERATION 2 out of 2 iterations

55.09   ###TreeTime.run: FINAL ROUND - confidence estimation via marginal
        reconstruction
Inferred a time resolved phylogeny using TreeTime:
    Sagulenko et al. TreeTime: Maximum-likelihood phylodynamic analysis
    Virus Evolution, vol 4, https://academic.oup.com/ve/article/4/1/vex042/4794731

updated tree written to trees/b117_seq_complete_sampUK_aligned_filt_mask_timetree.tre
node attributes written to refine.node.json</code></pre>
<p>As output, the file <code>output-tree trees/b117_seq_complete_sampUK_aligned_filt_mask_timetree.tre</code> will contain the time tree while the file <code>trees/refine.node.json</code> contains additional information about the tree. If you want, download the time tree and view it in <a href="https://icytree.org/">icytree</a>.</p>
<div id="optional-1" class="section level3 hasAnchor">
<h3>Optional<a href="maximum-likelihood-phylodynamic-analysis.html#optional-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you have spare time at the end of the lab, consider returning to conduct a sensitivity analysis on the specified clock rate and variance.</p>
<hr />
</div>
</div>
<div id="infer-ancestral-states-for-host-and-location-augur-traits" class="section level2 hasAnchor">
<h2>4.7. Infer ancestral states for host and location (<code>augur traits</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#infer-ancestral-states-for-host-and-location-augur-traits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will now be using treetime’s <code>mugration</code> model to reconstruct discrete ancestral states. In other words, we are trying to use the tree and genome metadata to reconstruct the most likely country at each of the internal nodes in the tree. It is called a mugration model because it is fundamentally estimating migration between “demes” (in this case, countries) as if it were a substitution rate matrix, commonly referred to as <em>mu</em>. See <a href="https://pubmed.ncbi.nlm.nih.gov/19779555/">Lemey et al. 2009</a> if you’re interested in learning more.</p>
<p>Side bar: discrete phylogeography, ancestral state reconstruction, ancestral trait inference, and ancestral character estimation can all be used interchangeably in this context.</p>
<p>As with the other analyses, augur makes this process very convenient:</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb88"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb88-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb88-1" tabindex="-1"></a><span class="ex">augur</span> traits <span class="at">--tree</span> trees/b117_seq_complete_sampUK_aligned_filt_mask_timetree.tre <span class="at">--metadata</span> data/masked/b117_seq_complete_sampUK_aligned_filt_mask.tsv <span class="at">--columns</span> Country Geo_Location <span class="at">--output-node-data</span> trees/trait.node.json</span></code></pre></div>
<p>The parameters we used are:</p>
<ul>
<li><code>--tree</code>: The time-calibrated tree we inferred in step 4</li>
<li><code>--metadata</code>: The metadata file with information about the genomes in the tree</li>
<li><code>--columns</code>: The columns in the metadata file we want to infer ancestral states for, in this case country and geo_location (more specific info).</li>
<li><code>--output-node-data</code>: Augur stores the additional information related to the ancestral trait inference for later visualisation.</li>
</ul>
<p>You should expect to see the following as output:</p>
<p><strong>Output</strong></p>
<pre><code>augur traits is using TreeTime version 0.11.4
Assigned discrete traits to 178 out of 178 taxa.

Assigned discrete traits to 178 out of 178 taxa.

Inferred ancestral states of discrete character using TreeTime:
    Sagulenko et al. TreeTime: Maximum-likelihood phylodynamic analysis
    Virus Evolution, vol 4, https://academic.oup.com/ve/article/4/1/vex042/4794731

results written to trait.node.json</code></pre>
<hr />
</div>
<div id="package-up-data-for-auspice-visualisation-augur-export" class="section level2 hasAnchor">
<h2>4.8. Package up data for Auspice visualisation (<code>augur export</code>)<a href="maximum-likelihood-phylodynamic-analysis.html#package-up-data-for-auspice-visualisation-augur-export" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will be using <a href="https://auspice.us/">Auspice</a> to visualize the tree alongside our metadata. To do this, we need to package up all of the data we have so far into a special file which can be used by Auspice. To do this, please run the following command:</p>
<p><strong>Commands</strong></p>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="maximum-likelihood-phylodynamic-analysis.html#cb90-1" tabindex="-1"></a><span class="ex">augur</span> export v2 <span class="at">--tree</span> trees/b117_seq_complete_sampUK_aligned_filt_mask_timetree.tre <span class="at">--node-data</span> trees/refine.node.json trees/trait.node.json <span class="at">--maintainers</span> <span class="st">&quot;CBW-IDE-2025&quot;</span> <span class="at">--title</span> <span class="st">&quot;Module 5 Practical&quot;</span> <span class="at">--output</span> trees/analysis-package.json <span class="at">--geo-resolutions</span> Country</span></code></pre></div>
<p>The parameters:</p>
<ul>
<li><code>--tree</code>: The time-calibrated tree</li>
<li><code>--node-data refine.node.json trait.node.json</code>: The 2 jsons created during the temporal and ancestral trait inference</li>
<li><code>--title "Module 5 Practical"</code>: A title for the auspice page, you can make this anything you want.</li>
<li><code>--maintainers "CBW-IDE-2025"</code>: A name for who is responsible for this analysis, useful for later, can make this anything you want.</li>
<li><code>--geo-resolutions Country</code>: Spatial resolution on which to plot the data. In this case we want country</li>
<li><code>--output analysis-package.json</code>: The key output file we want to generate. The file <code>trees/analysis-package.json</code> contains the tree with different branch length units (time and substitutions), our inferred ancestral traits, as well as additional data.</li>
</ul>
<p>You should expect to see the following as output:</p>
<p><strong>Output</strong></p>
<pre><code>Trait &#39;Geo_Location&#39; was guessed as being type &#39;categorical&#39;. Use a &#39;config&#39; file if you&#39;d like to set this yourself.
Trait &#39;Country&#39; was guessed as being type &#39;categorical&#39;. Use a &#39;config&#39; file if you&#39;d like to set this yourself.
Validating produced JSON
Validating schema of &#39;trees/analysis-package.json&#39;...
Validating that the JSON is internally consistent...
Validation of &#39;trees/analysis-package.json&#39; succeeded.</code></pre>
<hr />
<p><a name="visualize-tree"></a></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exercise-setup.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualizing-the-phylodynamic-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/bioinformaticsdotca//blob/main/050-module-5.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
